File: frontend/types/wardrobe.ts
export type BoundingBox = {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
};

export type ItemTags = {
  colors: string[];
  pattern?: string;
  style: string[];
  season: string[];
  occasion: string[];
};

export type WardrobeItem = {
  itemId: string;
  category: string;
  bbox: BoundingBox;
  imageUrl: string;
  tags: ItemTags;
};

File: backend/app/schemas/wardrobe.py
from pydantic import BaseModel, Field


class BoundingBox(BaseModel):
    x1: int
    y1: int
    x2: int
    y2: int


class ItemTags(BaseModel):
    colors: list[str] = Field(default_factory=list)
    pattern: str | None = None
    style: list[str] = Field(default_factory=list)
    season: list[str] = Field(default_factory=list)
    occasion: list[str] = Field(default_factory=list)


class WardrobeItemResponse(BaseModel):
    item_id: str
    category: str
    bbox: BoundingBox
    image_url: str
    tags: ItemTags
